tests/unit_tests/simpll/DifferentialFunctionComparatorTest.cpp

```diff
@@ -0,0 +1,1159 @@
+                0,
+                DScoL,
+                "test",
+                false,
+                "",
+                0,
+                "test",
+                DICompileUnit::DebugEmissionKind::FullDebug,
+                DTyArrL,
+                DIScopeArray{},
+                DIGlobalVariableExpressionArray{},
+                DIImportedEntityArray{},
+                DMacArrL,
+                0,
+                false,
+                false,
+                DICompileUnit::DebugNameTableKind::Default,
+                0);
+        DICompileUnit *DCUR = DICompileUnit::getDistinct(
+                CtxR,
+                0,
+                DScoR,
+                "test",
+                false,
+                "",
+                0,
+                "test",
+                DICompileUnit::DebugEmissionKind::FullDebug,
+                DTyArrR,
+                DIScopeArray{},
+                DIGlobalVariableExpressionArray{},
+                DIImportedEntityArray{},
+                DMacArrR,
+                0,
+                false,
+                false,
+                DICompileUnit::DebugNameTableKind::Default,
+                0);
+        DSubL = DISubprogram::get(CtxL,
+                                  DScoL,
+                                  "test",
+                                  "test",
+                                  DScoL,
+                                  1,
+                                  nullptr,
+                                  1,
+                                  nullptr,
+                                  0,
+                                  0,
+                                  DINode::DIFlags{},
+                                  DISubprogram::DISPFlags{},
+                                  DCUL);
+        DSubR = DISubprogram::get(CtxR,
+                                  DScoR,
+                                  "test",
+                                  "test",
+                                  DScoR,
+                                  1,
+                                  nullptr,
+                                  1,
+                                  nullptr,
+                                  0,
+                                  0,
+                                  DINode::DIFlags{},
+                                  DISubprogram::DISPFlags{},
+                                  DCUR);
+    }
+
+    /// Creates an enviroment for cmpGlobalValues to be able to compare
+    /// functions and then calls it via cmpBasicBlocks.
+    int testFunctionComparison(Function *FunL, Function *FunR) {
+        const std::string auxFunName = "AuxFunComp";
+
+        // Testing function comparison is a little bit tricky, because for the
+        // callee generation the call location must be set at the time the
+        // comparison is done.
+        // To ensure this a pair of auxilliary functions containing a call to
+        // the functions is added, along with their locations.
```

Comment:
This comment is quite hard to understand, I did not get it until after many readings. It should be at least improved so that it explains better why function comparison is done in such a complicated way. The comment should say that the functions `FunL` and `FunR` cannot be compared directly since `DifferentialFunctionComparator` expects a single pair of functions to compare (and such have already been provided in the constructor - `FL` and `FR`) and should another pair need to be compared, a new comparator must be created.
Anyway, why not do exactly that instead of doing this complicated stuff with function calls? Why not create a new (temporary) `TestComparator` object that will compare `FunL` and `FunR`?

Reply:
The point of `testFunctionComparison` isn't to test the comparison of `FunL` and `FunR` using `DifferentialFunctionComparator`'s `compare` method (this is unnecessary because the original method from `FunctionComparator` is used, which has already been tested in LLVM unit tests). 

The function is used for testing the part of `cmpGlobalValues` that compares functions (i.e. it checks whether it returns the correct return value and whether it calls `ModComparator->compareFunctions` properly). This part uses the private variables `CurrentLocL` and `CurrentLocR` that are set in `cmpBasicBlocks` based on the debug location of the instructions (call instructions in this case), hence the creation of the calls, their debug locations and using `cmpBasicBlocks` to get the result.

Reply:
I see, ok, then the comment should say this. Also, it is a bit strange that this is implemented as a class method while all other tests are implemented as standard GTest tests. The reason that I see here is that this function is used several times, but the function comment should probably state that.